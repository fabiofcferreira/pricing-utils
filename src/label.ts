export function formatToLabel({
  amount,
  currency,
  locale = 'en-GB',
}: {
  amount: number;
  currency: string;
  locale?: string;
}): string {
  return Intl.NumberFormat(locale, {
    style: 'currency',
    currency,
    currencyDisplay: 'symbol',
    currencySign: 'standard',
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  }).format(amount);
}
