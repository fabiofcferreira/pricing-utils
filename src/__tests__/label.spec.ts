import { formatToLabel } from '../label';

describe('Labels', () => {
  it('should generate correct labels for en-GB locale', () => {
    expect(
      formatToLabel({
        amount: 10.55,
        currency: 'GBP',
      }),
    ).toMatchSnapshot();

    expect(
      formatToLabel({
        amount: 10.2,
        currency: 'EUR',
      }),
    ).toMatchSnapshot();
  });

  it('should generate correct labels for en-US locale', () => {
    const locale = 'en-US';

    expect(
      formatToLabel({
        amount: 10.55,
        currency: 'GBP',
        locale,
      }),
    ).toMatchSnapshot();

    expect(
      formatToLabel({
        amount: 10.2,
        currency: 'EUR',
        locale,
      }),
    ).toMatchSnapshot();
  });

  it('should generate correct labels for pt-PT locale', () => {
    const locale = 'pt-PT';

    expect(
      formatToLabel({
        amount: 10.55,
        currency: 'GBP',
        locale,
      }),
    ).toMatchSnapshot();

    expect(
      formatToLabel({
        amount: 10.2,
        currency: 'EUR',
        locale,
      }),
    ).toMatchSnapshot();
  });
});
