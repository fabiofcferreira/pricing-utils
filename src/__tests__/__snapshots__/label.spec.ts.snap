// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labels should generate correct labels for en-GB locale 1`] = `"£10.55"`;

exports[`Labels should generate correct labels for en-GB locale 2`] = `"€10.20"`;

exports[`Labels should generate correct labels for en-US locale 1`] = `"£10.55"`;

exports[`Labels should generate correct labels for en-US locale 2`] = `"€10.20"`;

exports[`Labels should generate correct labels for pt-PT locale 1`] = `"10,55 £"`;

exports[`Labels should generate correct labels for pt-PT locale 2`] = `"10,20 €"`;
